src/pandas_profiling/model/describe.py

```diff
@@ -44,16 +50,39 @@ def describe(
     if df is None:
         raise ValueError("Can not describe a `lazy` ProfileReport without a DataFrame.")
 
-    if not isinstance(df, pd.DataFrame):
-        warnings.warn("df is not of type pandas.DataFrame")
+    # check for unwrapped dataframes and warn
+    if not isinstance(df, GenericDataFrame):
+        warnings.warn(UNWRAPPED_DATAFRAME_WARNING)
+        df_wrapper = get_appropriate_wrapper(df)
+        df = df_wrapper.preprocess(df)
+        df = df_wrapper(df)
 
     if df.empty:
         raise ValueError("df can not be empty")
 
+    if isinstance(df, SparkDataFrame):
+        # test if the version pyspark and pyarrow versions are compatible
+        test_for_pyspark_pyarrow_incompatibility()
+
+        # save the dataframe to speed up compute time
+
+        if config["spark"]["persist"]:
+            df.persist_bool = config["spark"]["persist"]
+            df.persist()
+        else:
+            df.persist_bool = False
+
     disable_progress_bar = not config["progress_bar"].get(bool)
 
     date_start = datetime.utcnow()
 
+    # if there are specific config for each correlation, then use those
+
+    if isinstance(df, SparkDataFrame):
```

Comment:
Obsolte if we use the change proposed above

Reply:
removed as obsolete